### PR TITLE
Fix configuration error display

### PR DESCRIPTION
**Pull request by phryneas**

### CODE REVIEW

File: src/language-server/config/config.ts [hunk 1]
```diff
@@ -197,24 +197,20 @@ export function parseApolloConfig(
...
-        unionSeparator: "\n  or\n",
-      }).toString(),
-    );
-    return null;
+    throw fromZodError(parsed.error, {
+      prefix: `Error parsing config file ${configURI?.fsPath}:`,
+      prefixSeparator: "\n",
+      issueSeparator: ";\n",
+      unionSeparator: "\n  or\n",
+    });
```
Comment (phryneas) [inline src/language-server/config/config.ts:203]: There was a full mechanism for handling thrown errors, but instead a lot of places logged errors instead. Defaulting to throwing in most places now.

File: src/language-server/server.ts [hunk 2]
```diff
@@ -19,9 +19,17 @@ import {
...
 
 const connection = createConnection(ProposedFeatures.all);
+
 Debug.SetConnection(connection);
+const { sendNotification: originalSendNotification } = connection;
+connection.sendNotification = async (...args: [any, ...any[]]) => {
+  await whenConnectionInitialized;
+  connection.sendNotification = originalSendNotification;
+  connection.sendNotification(...args);
+};
```
Comment (phryneas) [inline src/language-server/server.ts:32]: All `sendNotification` calls before the LSP client was fully started were swallowed previously - this included `Debug.error` etc.
This queues them instead.

File: src/language-server/server.ts [hunk 3]
```diff
@@ -61,14 +69,15 @@ workspace.onSchemaTags((params) => {
 });
 
 workspace.onConfigFilesFound(async (params) => {
-  await whenConnectionInitialized;
-
```
Comment (phryneas) [inline src/language-server/server.ts:65]: This was the one place that already had handling for that - we don't need that exception anymore.

**Comment by phryneas** [comment]
No ideas why tests are failing on this, they pass locally. Gotta take a look tomorrow.

**Comment by phryneas** [comment]
So... `--runInBand` fixes this. 🤷

**Review by alessbell — APPROVED** [review]

**Pull request merged**